rest-api/src/main/java/gov/cms/qpp/conversion/api/services/DbServiceImpl.java

```diff
@@ -59,15 +63,20 @@
 	 * @return {@link List} of unprocessed {@link Metadata}
 	 */
 	public List<Metadata> getUnprocessedCpcPlusMetaData() {
-		HashMap<String, AttributeValue> valueMap = new HashMap<>();
-		valueMap.put(":cpcValue", new AttributeValue().withN("1"));
-		valueMap.put(":cpcProcessedValue", new AttributeValue().withN("0"));
 
-		DynamoDBScanExpression metadataScan = new DynamoDBScanExpression()
-				.withFilterExpression("Cpc = :cpcValue and CpcProcessed = :cpcProcessedValue")
-				.withExpressionAttributeValues(valueMap);
+		return IntStream.range(0, Constants.CPC_DYNAMO_PARTITIONS).mapToObj(partition -> {
+			HashMap<String, AttributeValue> valueMap = new HashMap<>();
+			valueMap.put(":cpcValue", new AttributeValue().withS(Constants.CPC_DYNAMO_PARTITION_START + partition));
+			valueMap.put(":cpcProcessedValue", new AttributeValue().withS("false"));
+
+			DynamoDBQueryExpression<Metadata> metadataQuery = new DynamoDBQueryExpression<Metadata>()
+				.withIndexName("Cpc-CpcProcessedCreateDate-index")
+				.withKeyConditionExpression("Cpc = :cpcValue and begins_with(CpcProcessedCreateDate, :cpcProcessedValue)")
+				.withExpressionAttributeValues(valueMap)
+				.withConsistentRead(false);
 
-		return mapper.scan(Metadata.class, metadataScan);
+			return mapper.query(Metadata.class, metadataQuery).stream();
+		}).flatMap(Function.identity()).collect(Collectors.toList());
```

Comment:
what happens with flatMap(Function.identity()) -- wouldn't that just be (functionally) the same stream? am I misunderstanding that?

Reply:
While `Function.identity()` does return the same stream, `flatMap` "unwraps" all the streams returned from `mapper.query(Metadata.class, metadataQuery).stream()`.  This is all because I want a `List<Metadata>` and not a `List<List<Metadata>>`.

Reply:
I think

```
			return mapper.query(Metadata.class, metadataQuery);
		}).flatMap(Collection::stream).collect(Collectors.toList());
```

would be a little more clear. Thoughts?